custom_components/mammotion/lawn_mower.py

```diff
@@ -85,71 +84,110 @@ async def async_start_mowing(self) -> None:
         """Start mowing."""
         # check if job in progress
         #
-        if self.rpt_dev_status is None:
+        mode = self.rpt_dev_status.sys_status
+        if mode is None:
             raise HomeAssistantError(
                 translation_domain=DOMAIN, translation_key="device_not_ready"
             )
-        work_area = self.report_data.work.area >> 16
-
-        if work_area > 0 and (
-            self.rpt_dev_status.sys_status == WorkMode.MODE_PAUSE
-            or self.rpt_dev_status.sys_status == WorkMode.MODE_READY
+        
+        if (
+            mode == WorkMode.MODE_WORKING
+            or mode == WorkMode.MODE_PAUSE
+            or mode == WorkMode.MODE_READY
+            or mode == WorkMode.MODE_RETURNING
         ):
             try:
-                await self.coordinator.async_send_command("resume_execute_task")
-                return await self.coordinator.async_request_iot_sync()
+                if mode == WorkMode.MODE_RETURNING:
+                    trans_key = "dock_failed"
+                    await self.coordinator.async_send_command("cancel_return_to_dock")
+                    await self.coordinator.async_request_iot_sync()
+                    mode = self.rpt_dev_status.sys_status
+                if mode == WorkMode.MODE_PAUSE :
+                    trans_key = "resume_failed"
```

Comment:
If the mower is on the charger recharging its in WorkMode.MODE_READY but needs resumed, for example I resume mine early sometimes at 80% charge

Reply:
Mine is definitely in "MODE_PAUSE" when recharging.
![image](https://github.com/user-attachments/assets/e710f590-6e49-4e1f-a7f1-30df219f0c5d)

Reply:
Conversely work area does not update to "0" when task is cancelled, if that is what "area" was checking. 

![image](https://github.com/user-attachments/assets/098aabf9-219a-4588-9b6c-f70a04178a47)